File: back_api/acessibilityguide/src/main/java/com/acessibilityguide/api/config/EbeanFactoryBean.java
package com.acessibilityguide.api.config;

import javax.sql.DataSource;

import org.springframework.beans.factory.FactoryBean;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import com.avaje.ebean.EbeanServer;
import com.avaje.ebean.EbeanServerFactory;
import com.avaje.ebean.config.ServerConfig;
import com.avaje.ebean.springsupport.txn.SpringAwareJdbcTransactionManager;

@Component
public class EbeanFactoryBean implements FactoryBean<EbeanServer> {

	@Autowired
	private DataSource dataSource;
	
	public final static String MODELS_PACKAGE="com.acessibilityguide.api.model";
	
	@Override
	public EbeanServer getObject() throws Exception {
		ServerConfig config = new ServerConfig();
		config.setName("duo_db");
		config.setDataSource(dataSource);
		
		config.setExternalTransactionManager(new SpringAwareJdbcTransactionManager());
		config.loadFromProperties();
		
		config.setDefaultServer(true);
		config.setRegister(true);
		config.addPackage(MODELS_PACKAGE);		
		
		return EbeanServerFactory.create(config);
	}

	@Override
	public Class<?> getObjectType() {
		return EbeanServer.class;
	}

	@Override
	public boolean isSingleton() {
		return true;
	}
	
}

File: back_api/acessibilityguide_back/src/main/kotlin/com/acessibilityguide/api/config/EbeanFactoryBean.kt
package com.acessibilityguide.api.config

import com.avaje.ebean.EbeanServer
import org.springframework.beans.factory.FactoryBean
import com.avaje.ebean.EbeanServerFactory
import com.avaje.ebean.springsupport.txn.SpringAwareJdbcTransactionManager
import com.avaje.ebean.config.ServerConfig
import org.springframework.beans.factory.annotation.Autowired
import javax.sql.DataSource


/**
 * Created by eduardolopes on 06/06/17.
 */

class EbeanFactoryBean: FactoryBean<EbeanServer> {

    @Autowired
    private val dataSource: DataSource? = null

    val MODELS_PACKAGE = "com.acessibilityguide.api.model"

    override fun getObject(): EbeanServer {
        val config = ServerConfig()
        config.name = "db"
        config.dataSource = dataSource

        config.externalTransactionManager = SpringAwareJdbcTransactionManager()
        config.loadFromProperties()

        config.isDefaultServer = true
        config.isRegister = true
        config.addPackage(MODELS_PACKAGE)

        return EbeanServerFactory.create(config)
    }

    override fun isSingleton(): Boolean {
        return true
    }

    override fun getObjectType(): Class<*> {
        return EbeanServer::class.java;
    }
}